src/campaign_server/campaign_server.cpp

```diff
@@ -752,14 +799,14 @@ void server::handle_delete(const server::request& req)
 
 	LOG_CS << "deleting campaign '" << erase["name"] << "' requested from " << req.addr << "\n";
 
-	const config& campaign = get_campaign(erase["name"]);
+	config& campaign = get_campaign(erase["name"]);
```

Comment:
A little bit of stylistic pedantry that does not actually need to be corrected (if I understand correctly, see at the end):

I see that making ``campaign`` non-const is necessary because ``authenticate()`` takes a non-const config reference. Why does it need to do this, though, when ``util::create_hash()`` takes const references?

Do note that ``config::operator[]`` (as used in ``authenticate()``) has different semantics with missing attributes depending on whether the non-const version is called or not: the const version returns a reference to a dummy blank attribute, while the non-const version auto-creates the missing attribute and returns a reference to it. In practical terms, this means that ``authenticate()`` will write into the server config an empty ``passsalt`` or ``passhash`` in the event that they don't already exist (which I understand should not normally happen since add-on entries using plain-text passphrases get converted on config load).